components/suggest/src/store.rs

```diff
@@ -460,13 +409,469 @@ mod tests {
+                        },
+                    }, {
+                        "id": "icon-2",
+                        "type": "icon",
+                        "last_modified": 20,
+                        "attachment": {
+                            "filename": "icon-2.png",
+                            "mimetype": "image/png",
+                            "location": "icon-2.png",
+                            "hash": "",
+                            "size": 0,
+                        },
+                    }],
+                }))?)
+            }
+
+            fn suggestions_per_data_attachment(&self) -> u64 {
+                unreachable!(
+                    "Didn't expect test to ask for number of suggestions per data attachment"
+                )
+            }
+
+            fn get_data_attachment(
+                &self,
+                location: &str,
+            ) -> Result<DownloadedSuggestDataAttachment> {
+                Ok(serde_json::from_value(match location {
+                    "data-1.json" => json!([{
+                        "id": 0,
+                        "advertiser": "Good Place Eats",
+                        "iab_category": "8 - Food & Drink",
+                        "keywords": ["la", "las", "lasa", "lasagna", "lasagna come out tomorrow"],
+                        "title": "Lasagna Come Out Tomorrow",
+                        "url": "https://www.lasagna.restaurant",
+                        "icon": "2",
+                    }, {
+                        "id": 0,
+                        "advertiser": "Good Place Eats",
+                        "iab_category": "8 - Food & Drink",
+                        "keywords": ["pe", "pen", "penne", "penne for your thoughts"],
+                        "title": "Penne for Your Thoughts",
+                        "url": "https://penne.biz",
+                        "icon": "2",
+                    }]),
+                    _ => unreachable!(
+                        "Didn't expect test to request data attachment `{}`",
+                        location
+                    ),
+                })?)
+            }
+
+            fn get_icon_attachment(&self, location: &str) -> Result<Vec<u8>> {
+                Ok(match location {
+                    "icon-2.png" => "i-am-an-icon".as_bytes().into(),
+                    _ => unreachable!(
+                        "Didn't expect test to request icon attachment `{}`",
+                        location
+                    ),
+                })
+            }
+        }
+
+        let store = GenericSuggestStore::new(
+            "file:ingest_icons?mode=memory&cache=shared",
+            TestSettingsClient,
+        );
+        store.ingest(SuggestIngestionConstraints::default())?;
+
+        assert_eq!(
+            store.query(SuggestionQuery {
+                keyword: "la".into(),
+                include_sponsored: true,
+                ..Default::default()
+            })?,
+            &[Suggestion {
```

Comment:
Looks like we have to write lots of those fixture records in the assertions throughout the file, shall we consider using a snapshot test crate (e.g. [insta](https://github.com/mitsuhiko/insta)) to ease that?

Reply:
I added `expect-test` to get us started with snapshot testing, and I'm such a fan! Thanks for the pointer!